avaframe/ana5Utils/distanceTimeAnalysis.py

```diff
@@ -0,0 +1,792 @@
+    # fetch info on distance of runout area start point index in s array
+    indStartOfRunout = rasterTransfo['indStartOfRunout']
+    # create range raster with respect to runout area start point
+    # minus in upstream direction of runout area start point
+    rangeGates = rasterTransfo['s'][:] - rasterTransfo['s'][indStartOfRunout]
+    rangeRaster = np.repeat([rangeGates], newRasterDEM.shape[1], axis=0).T
+
+    # create dictionary with info on front distance, mean values, range distance array
+    mtiInfo = {'mti': '', 'rangeList': [], 'timeList': []}
+    # mtiInfo['rangeGates'] = rasterTransfo['s'][indStartOfRunout] - rasterTransfo['s'][:]
+    mtiInfo['rangeGates'] = rangeGates
+    mtiInfo['betaPoint'] = [rasterTransfo['x'][indStartOfRunout], rasterTransfo['y'][indStartOfRunout]]
+    mtiInfo['betaPointAngle'] = rasterTransfo['startOfRunoutAreaAngle']
+    mtiInfo['rangeRaster'] = rangeRaster
+    mtiInfo['rasterTransfo'] = rasterTransfo
+    mtiInfo['type'] = 'thalwegTime'
+    mtiInfo['z'] = rasterTransfo['z']
+    mtiInfo['s'] = rasterTransfo['s']
+    mtiInfo['referencePointName'] = 'beta point'
+
+    return mtiInfo
+
+
+def extractFrontAndMeanValues(cfgRangeTime, flowF, demHeader, mtiInfo):
+    """ extract avalanche front and mean values of flow parameter result field
+        used for colorcoding range-time diagram
+
+        Parameters
+        ------------
+        cfgRangeTime: configparser object
+            configuration settings
+        flowF: numpy nd array
+            flow parameter result field
+        demHeader: dict
+            dictionary with info on DEM header used for locating fields
+        mtiInfo: dict
+            dictionary with arrays, lists for range-time diagram:
+            mti for colorcoding (average values of range gates/ crossprofiles)
+            timeList time step info
+            rangeList range info
+            rangeGates info on cross profile locations
+            rangeRaster - info on distance from beta point in path following coordinate system
+            rasterTransfo - information on domain transformation to path following coordinate system
+
+        Returns
+        --------
+        mtiInfo: dict
+            updated dictionary with info on average value along crossprofiles (mti)
+            and distance to front from reference point along path (rangeList)
+    """
+
+    # setup input parameters
+    rasterTransfo = mtiInfo['rasterTransfo']
+
+    # add header to result field - required for coordinate transformation
+    fieldFull = {'header': demHeader, 'rasterData': flowF}
+
+    # transorm result field into avalanche path following coordinate system
+    slRaster = aT.transform(fieldFull, rasterTransfo, cfgRangeTime['GENERAL']['interpMethod'])
+
+    # fetch raster area and compute mean, max values for each cross-profile
+    # TODO: average over cells – weighted will cell area but only projected area (aimec function)
```

Comment:
Address todo right away?

Reply:
made an issue as interesting for aimec too: https://github.com/avaframe/AvaFrame/issues/695